### ISSUE TITLE
Undistribute a distributed table

### ISSUE THREAD

**Issue by hanefi** [citusdata/citus#3804]
After creating a distributed table, it is not straight-forward to undo the distribution and revert the table back to being a local one. Provide a mechanism t undistribute a table. Here is an idea:

```SQL
CREATE TABLE customers(id uuid, name text, data jsonb);
SELECT create_distributed_table('customers', 'name');

-- realize that id is a better distribution column for application needs
SELECT undistribute_table('customers');
SELECT create_distributed_table('customers', 'id');
```

## Some of the benefits:
1. If the user had some writes on the distributed table and they decide to drop the shards, they will loose the updates on the distributed table as the local records will be outdated.
Related https://github.com/citusdata/citus/issues/1770#issuecomment-446205988 :

2. This will make auto truncates after distributing tables easier. (See #2624) One of the main reasons why we keep the local records is that we do not have an easy mechanism to pull all the data in the shards to the original local table. Once we provide that functionality, it might be easier to truncate local shards

## Current Workaround
I think we have several options, and here is one of the simplest workarounds:

i. Create a temporary table with all the records in all the shards of the distributed table. 
`CREATE TEMPORARY TABLE customers_temp AS SELECT * FROM customers`

ii. Drop the distributed table
`DROP TABLE customers`

iii. Recreate the local table using the temporary table
`CREATE TABLE customers AS SELECT * FROM customers_temp`

If we have foreign keys however, it is not as easy to undistribute the table.

**Comment by onurctirtir** [comment]
> iii. Recreate the local table using the temporary table
CREATE TABLE customers AS SELECT * FROM customers_temp

CREATE TABLE AS doesn't cover all the objects that a table has. We should probably find a way to pull the data from the shards to the table itself (though this could be hard to do), we shouldn't create a new table with `CREATE TABLE AS`

**Comment by onurctirtir** [comment]
> If we have foreign keys however, it is not as easy to undistribute the table.

Temporarily marking constraints on the as invalid, then pulling back the data, then marking those constrains to be valid again could be a solution.

**Comment by metdos** [comment]
Same as #2334

**Comment by metdos** [comment]
Fixed with #3913
